### PR TITLE
AI UI library

### PR DESCRIPTION
**Pull request by yschimke**
#### WHAT

AI UI library

#### WHY

Show some samples of AI such as Gemini on Wear.

#### Checklist :clipboard:
- [x] Add explicit visibility modifier and explicit return types for public declarations
- [x] Run spotless check
- [x] Run tests
- [x] Update metalava's signature text files

### CODE REVIEW

**Comment by amankhoza** [comment]
LGTM

File: ai/ui/src/main/java/com/google/android/horologist/ai/ui/components/PromptDisplay.kt [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+public fun TextPromptDisplay(
+    prompt: TextPromptUiModel,
+    modifier: Modifier = Modifier,
+    onClick: () -> Unit = {},
+) {
+    Card(
+        modifier = modifier.fillMaxWidth(),
+        onClick = onClick,
+        backgroundPainter = CardDefaults.cardBackgroundPainter(
+            Color(0xff8ab4f8),
```
Comment (fstanis) [inline ai/ui/src/main/java/com/google/android/horologist/ai/ui/components/PromptDisplay.kt:46]: Maybe extract this color into a named `const`?

File: ai/ui/src/main/java/com/google/android/horologist/ai/ui/model/PromptUiModel.kt [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.android.horologist.ai.ui.model
+
+import java.time.Instant
+
+public sealed interface PromptUiModel : PromptResponseUiModel
```
Comment (fstanis) [inline ai/ui/src/main/java/com/google/android/horologist/ai/ui/model/PromptUiModel.kt:21]: Maybe I'm missing something, but prompt being a subclass of prompt response seems a bit odd, is this international?

Comment (yschimke) [reply ai/ui/src/main/java/com/google/android/horologist/ai/ui/model/PromptUiModel.kt:21]: PromptResponse means PromptOrResponse.  I'll rename.

Comment (yschimke) [reply ai/ui/src/main/java/com/google/android/horologist/ai/ui/model/PromptUiModel.kt:21]: Will follow up with this.

File: ai/ui/src/main/java/com/google/android/horologist/ai/ui/screens/PromptScreen.kt [hunk 3]
```diff
@@ -0,0 +1,108 @@
...
+    promptEntry: @Composable () -> Unit,
+) {
+    ScalingLazyColumn(columnState = columnState, modifier = modifier) {
+        item {
+            ListHeader(modifier = Modifier.fillMaxWidth(0.8f)) {
+                Text(text = uiState.modelInfo?.name ?: "...")
+            }
+        }
+        uiState.messages.forEach {
+            item {
```
Comment (fstanis) [inline ai/ui/src/main/java/com/google/android/horologist/ai/ui/screens/PromptScreen.kt:65]: Is there something we can use as a key here? May not be needed yet, just thinking about future-proofing.

**Review by fstanis — APPROVED** [review]

File: ai/ui/src/main/java/com/google/android/horologist/ai/ui/screens/PromptScreen.kt (hunk 3 above)
Comment (yschimke) [reply ai/ui/src/main/java/com/google/android/horologist/ai/ui/screens/PromptScreen.kt:65]: Good question.  Not sure.

Also a bigger question of how big these will get.  Should they use paging to lazy load them?

Not going to think about this yet.

File: ai/ui/src/main/java/com/google/android/horologist/ai/ui/components/PromptDisplay.kt (hunk 1 above)
Comment (yschimke) [reply ai/ui/src/main/java/com/google/android/horologist/ai/ui/components/PromptDisplay.kt:46]: I really don't think I typed that value.  wonder if refactoring replaced a constant or something....

```
public class Colors(
    primary: Color = Color(0xFFAECBFA),
    primaryVariant: Color = Color(0xFF8AB4F8),
```

Comment (yschimke) [reply ai/ui/src/main/java/com/google/android/horologist/ai/ui/components/PromptDisplay.kt:46]: Color can't be const can it? just a val. I'll change to MaterialTheme.colors.primaryVariant

**Pull request merged**